recipes/hyperscan/all/patches/fix-cmake.patch

```diff
@@ -220,35 +220,216 @@ index 8bc6077..579a7b4 100644
+         study_size = 0;
+     } else {
+-        if (pcre_fullinfo(compiled, extra, PCRE_INFO_STUDYSIZE, &study_size)) {
++        if (bridge_pcre_fullinfo(compiled, extra, PCRE_INFO_STUDYSIZE, &study_size)) {
+             throw CompileError(PCRE_ERROR_MSG);
+         }
+     }
+ 
+-    if (pcre_fullinfo(compiled, extra, PCRE_INFO_CAPTURECOUNT, &capture_cnt)) {
++    if (bridge_pcre_fullinfo(compiled, extra, PCRE_INFO_CAPTURECOUNT, &capture_cnt)) {
+         throw CompileError(PCRE_ERROR_MSG);
+     }
+ 
+@@ -377,7 +377,7 @@ void PatternData::buildPcre(const char *pattern, u32 flags) {
+      * even in the pure unguarded pcre mode where there is no hs available. We
+      * can not use the compile flags due to (*UTF8) verb */
+     unsigned long int opts = 0; // PCRE_INFO_OPTIONS demands an unsigned long
+-    if (pcre_fullinfo(compiled, extra, PCRE_INFO_OPTIONS, &opts)) {
++    if (bridge_pcre_fullinfo(compiled, extra, PCRE_INFO_OPTIONS, &opts)) {
+         throw CompileError(PCRE_ERROR_MSG);
+     }
+     utf8 = opts & PCRE_UTF8;
+diff --git a/chimera/ch_database.h b/chimera/ch_database.h
+index 28bde86..28459e8 100644
+--- a/chimera/ch_database.h
++++ b/chimera/ch_database.h
+@@ -38,9 +38,7 @@ extern "C"
  {
  #endif
  
 -#define PCRE_STATIC
+-#include <pcre.h>
+-
++#include "ch_pcre_bridge.h"
+ #include "ch_compile.h" // for CH_MODE_ flags
+ #include "ue2common.h"
+ #include "hs_version.h"
+diff --git a/chimera/ch_pcre_bridge.c b/chimera/ch_pcre_bridge.c
+new file mode 100644
+index 0000000..1ff8a84
+--- /dev/null
++++ b/chimera/ch_pcre_bridge.c
+@@ -0,0 +1,35 @@
++// HOTFIX TO PREVENT CALLING CONVENTION MISMATCH ON MSVC
++
++#include "ch_pcre_bridge.h"
++
++// Internal PCRE func.
++extern int _pcre_valid_utf(const unsigned char *, int, int *);
++
++pcre * bridge_pcre_compile(const char * pattern, int options, const char ** errptr, int * erroffset, const unsigned char * tableptr)
++{
++    return pcre_compile(pattern, options, errptr, erroffset, tableptr);
++}
++
++int bridge_pcre_fullinfo(const pcre *code, const pcre_extra *extra, int what, void *where)
++{
++    return pcre_fullinfo(code, extra, what, where);
++}
++
++pcre_extra * bridge_pcre_study(const pcre *code, int options, const char **errptr)
++{
++    return pcre_study(code, options, errptr);
++}
++
++int bridge_pcre_exec(const pcre *code, const pcre_extra *extra, const char *subject, int length, int startoffset, int options, int *ovector, int ovecsize)
++{
++    return pcre_exec(code, extra, subject, length, startoffset, options, ovector, ovecsize);
++}
++
++int bridge_pcre_valid_utf(const unsigned char *string, int length, int *erroroffset)
++{
++    return _pcre_valid_utf(string, length, erroroffset);
++}
++
++
++
++
+diff --git a/chimera/ch_pcre_bridge.h b/chimera/ch_pcre_bridge.h
+new file mode 100644
+index 0000000..bc78c7c
+--- /dev/null
++++ b/chimera/ch_pcre_bridge.h
+@@ -0,0 +1,32 @@
++#pragma once
++
++#ifndef CH_PCRE_BRIDGE_H_
++#define CH_PCRE_BRIDGE_H_
```

Comment:
An include guard and `pragma once`?

Reply:
Yes, it is usual way to do: ```pragma once``` before include guards for compatibility.